data/resolvers.js

```diff
@@ -2,11 +2,18 @@ import { Users } from './connectors';
 
 const resolvers = {
   Query: {
-    login(_, { username, password }) {
-      return new Promise((resolve, reject) => {
-        setTimeout( () => reject('MongoDB timeout when fetching user (timeout is 500ms)'), 500);
-        Users.findOne( { "username": username, "password": password } ).then(
-          (user) => resolve( { key: "somekey", role: "someRole", age: user.age, joinedAt: user.joinedAt }));
+    login: (_, { email, pw }) => {
+      return Users.findOne({ "email": email, "password": pw });
+    }
+  },
+  Mutation: {
+    createUser: (root, { email, pw }) => {
+      return Users.findOne({ "email": email, "password": pw }).then( (user) => {
+        if (user != null) return null
```

Comment:
As we are in JS world, should this not just be a falsy test: `if (!user) return null;`